plugin/trino-iceberg/src/main/java/io/airlift/compress/Compressor.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.compress;
+
+import java.nio.ByteBuffer;
+
+public interface Compressor
```

Comment:
This module is already pulling in the v1 APIs, so shouldn't this already be in the classpath?

Generally I think a better approach would be to patch the puffin code to have a compression interface we could replace.

Reply:
Yes that makes sense but requires a work there and I think we shouldn't wait but do this as a followup